blocks/bricks/base.py

```diff
@@ -646,7 +647,8 @@ def initialize(self):
         if not self.initialization_config_pushed:
             self.push_initialization_config()
         for child in self.children:
-            child.initialize()
+            if not child.initialized:
+                child.initialize()
```

Comment:
This is too radical of a solution. I can easily imagine that there is a lot of code where bricks are initialized, then initialization schemes are altered, then they are initialized again.

At the same time, I think it would be okay to do it for allocation, because one normally does not want to reallocate a brick.